src/transformers/pipelines/base.py

```diff
@@ -707,15 +767,42 @@ def ensure_tensor_on_device(self, **inputs):
         Ensure PyTorch tensors are on the specified device.
 
         Args:
-            inputs (keyword arguments that should be :obj:`torch.Tensor`): The tensors to place on :obj:`self.device`.
+            inputs (keyword arguments that should be :obj:`torch.Tensor`, the rest is ignored): The tensors to place on :obj:`self.device`.
+            Recursive on lists **only**.
 
         Return:
             :obj:`Dict[str, torch.Tensor]`: The same as :obj:`inputs` but on the proper device.
         """
-        return {
-            name: tensor.to(self.device) if isinstance(tensor, torch.Tensor) else tensor
-            for name, tensor in inputs.items()
-        }
+        return self._ensure_tensor_on_device(inputs)
+
+    def _ensure_tensor_on_device(self, inputs):
+        if isinstance(inputs, dict):
```

Comment:
why not:

```suggestion
        if isinstance(inputs, dict) or isinstance(inputs, list):
            return self.ensure_tensor_on_cpu(inputs)
```
?

Reply:
why do dict and list always have to be  on CPU? Why is that?
Also do we really need that function? It's not really intuitive for me that `ensure_tensor_on_device` puts all lists and dicts on CPU and only tensors to `self.device`. 

Couldn't we just have a:

```
        if isinstance(inputs, dict) or isinstance(inputs, list):
            return self.ensure_tensor_on_cpu(inputs)
        elif isinstance(inputs, torch.Tensor):
                    return inputs.to(self.device)
        else:
             return inputs
```
instead of having this function at all?

Reply:
Bad copy&past, it should just be recursive.